packages/platforms/browser/lib/auto-instrumentation/page-load-phase-spans.ts

```diff
@@ -0,0 +1,73 @@
+import type { SpanFactory } from '@bugsnag/core-performance'
+import { type PerformanceWithTiming } from '../on-settle/load-event-end-settler'
+
+function shouldOmitSpan (startTime: number, endTime: number) {
+  return startTime === 0 && endTime === 0
+}
+
+export const instrumentPageLoadPhaseSpans = (spanFactory: SpanFactory, route: string, performance: PerformanceWithTiming) => {
+  const entry = performance.getEntriesByType('navigation')[0] as PerformanceNavigationTiming
+
+  if (entry) {
+    if (!shouldOmitSpan(entry.unloadEventStart, entry.unloadEventEnd)) {
+      spanFactory.endSpan(spanFactory.startSpan('[PageLoadPhase/Unload]' + route, {
+        startTime: entry.unloadEventStart
+      }), entry.unloadEventEnd)
+    }
+
+    if (entry.redirectStart && !shouldOmitSpan(entry.redirectStart, entry.redirectEnd)) {
+      spanFactory.endSpan(spanFactory.startSpan('[PageLoadPhase/Redirect]' + route, {
+        startTime: entry.redirectStart
+      }), entry.redirectEnd)
+    }
+
+    if (!shouldOmitSpan(entry.fetchStart, entry.domainLookupStart)) {
+      spanFactory.endSpan(spanFactory.startSpan('[PageLoadPhase/LoadFromCache]' + route, {
+        startTime: entry.fetchStart
+      }), entry.domainLookupStart)
+    }
+
+    if (!shouldOmitSpan(entry.domainLookupStart, entry.domainLookupEnd)) {
+      spanFactory.endSpan(spanFactory.startSpan('[PageLoadPhase/DNSLookup]' + route, {
+        startTime: entry.domainLookupStart
+      }), entry.domainLookupEnd)
+    }
+
+    const TCPHandshakeEnd = Math.min(entry.connectEnd, entry.secureConnectionStart)
+    if (!shouldOmitSpan(entry.connectStart, TCPHandshakeEnd)) {
+      spanFactory.endSpan(spanFactory.startSpan('[PageLoadPhase/TCPHandshake]' + route, {
+        startTime: entry.connectStart
+      }), TCPHandshakeEnd)
+    }
```

Comment:
good point - and I guess if `secureConnectionStart` is > 0 it will always be smaller than `connectEnd`, so I guess we should just always use `secureConnectionStart` if it's greater than 0, otherwise use `connectEnd`?

```
const TCPHandshakeEnd = entry.secureConnectionStart || entry.connectStart
```